### PR TITLE
My code Review Suggestions

### PR DESCRIPTION
**Pull request by gormlyja**

### CODE REVIEW

File: app/src/main/java/edu/uc/app/parkfinder/RetrofitClientInstance.kt [hunk 1]
```diff
@@ -6,7 +6,7 @@ import retrofit2.converter.gson.GsonConverterFactory
 object RetrofitClientInstance {
 
     private var retrofit: Retrofit? = null
-    private val BASE_URL = "https://pkgstore.datahub.io"
+    private val BASE_URL = "https://pkgstore.datahub.io/core/country-list/data_json/data/8c458f2d15d9f2119654b29ede6e45b8/"
```
Comment (gormlyja) [inline app/src/main/java/edu/uc/app/parkfinder/RetrofitClientInstance.kt:9]: Made BASE_URL longer in order to make the @GET for retrofit more readable

File: app/src/main/java/edu/uc/app/parkfinder/dao/IParkDAO.kt [hunk 2]
```diff
@@ -6,6 +6,6 @@ import retrofit2.http.GET
 
 interface IParkDAO {
 
-    @GET("/core/country-list/data_json/data/8c458f2d15d9f2119654b29ede6e45b8/data_json.json")
+    @GET("data_json.json/")
```
Comment (gormlyja) [inline app/src/main/java/edu/uc/app/parkfinder/dao/IParkDAO.kt:9]: Shorter URL for readability

File: app/src/main/java/edu/uc/app/parkfinder/dto/Park.kt [hunk 3]
```diff
@@ -2,8 +2,6 @@ package edu.uc.app.parkfinder.dto
 
 import com.google.gson.annotations.SerializedName
 
-data class Park(@SerializedName("Code") var code: String,
-                @SerializedName("Name") var name: String) {
-
+data class Park(@SerializedName("Code") var code: String, @SerializedName("Name") var name: String) {
```
Comment (gormlyja) [inline app/src/main/java/edu/uc/app/parkfinder/dto/Park.kt:5]: Unless variable declarations take more than a screen of space, I would keep it to one line. For 2 variables, it would be acceptable and readable to leave in one line.

File: app/src/main/java/edu/uc/app/parkfinder/service/ParkService.kt [hunk 4]
```diff
@@ -11,7 +11,7 @@ import retrofit2.Response
 class ParkService {
 
     fun fetchParks() : MutableLiveData<ArrayList<Park>> {
-        var _parks = MutableLiveData<ArrayList<Park>>()
+        var parks = MutableLiveData<ArrayList<Park>>()
```
Comment (gormlyja) [inline app/src/main/java/edu/uc/app/parkfinder/service/ParkService.kt:14]: no _ before variable name. Follows naming convention

File: app/src/main/java/edu/uc/app/parkfinder/service/ParkService.kt [hunk 5]
```diff
@@ -33,10 +33,10 @@ class ParkService {
              */
             override fun onResponse(call: Call<ArrayList<Park>>, response: Response<ArrayList<Park>>
             ) {
-                _parks.value = response.body()
+                parks.value = response.body()
```
Comment (gormlyja) [inline app/src/main/java/edu/uc/app/parkfinder/service/ParkService.kt:36]: Follows initial variable name change

File: app/src/main/java/edu/uc/app/parkfinder/service/ParkService.kt [hunk 6]
```diff
@@ -33,10 +33,10 @@ class ParkService {
...
             }
         })
 
-        return _parks
+        return parks
```
Comment (gormlyja) [inline app/src/main/java/edu/uc/app/parkfinder/service/ParkService.kt:40]: Follows initial variable name change

File: app/src/main/java/edu/uc/app/parkfinder/ui/main/MainActivity.kt [hunk 7]
```diff
@@ -9,6 +9,7 @@ class MainActivity : AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.main_activity)
+
```
Comment (gormlyja) [inline app/src/main/java/edu/uc/app/parkfinder/ui/main/MainActivity.kt:12]: line break to keep code easily readable

File: app/src/main/java/edu/uc/app/parkfinder/ui/main/MainViewModel.kt [hunk 8]
```diff
@@ -13,7 +13,7 @@ class MainViewModel : ViewModel() {
         fetchParks()
     }
 
-    fun fetchParks() {
+    private fun fetchParks() {
```
Comment (gormlyja) [inline app/src/main/java/edu/uc/app/parkfinder/ui/main/MainViewModel.kt:16]: function called only from another in the same file, so I made this private

File: app/src/main/java/edu/uc/app/parkfinder/ui/main/SearchResultDirections.kt [hunk 9]
```diff
@@ -4,9 +4,9 @@ import androidx.navigation.ActionOnlyNavDirections
 import androidx.navigation.NavDirections
 import edu.uc.app.parkfinder.R
 
-public class SearchResultDirections private constructor() {
-  public companion object {
-    public fun actionMainFragmentToResultFragment(): NavDirections =
+class SearchResultDirections private constructor() {
+    companion object {
+      fun actionMainFragmentToResultFragment(): NavDirections =
```
Comment (gormlyja) [inline app/src/main/java/edu/uc/app/parkfinder/ui/main/SearchResultDirections.kt:9]: public modifier is redundant

**Comment by gormlyja** [comment]
Analysis of the program:
Overall, program is still in early development. Back end is still relying on a JSON other than the one that includes material related to the app. I could not find a JSON that would be usable here for my review, but it is imperative to find a JSON to base this on to make sure the app is finish-able to your desire. The front end is simple and makes sense, but maybe instead of trying to search by name, maybe search by city and the app pulls parks in the city. People will know where they are willing to travel for their park before they know the name of the park they want to go normally. I like the idea, just needs some more work to ensure confidence in its potential to truly shine as an app.

The program was available on time. 

I would consider commenting more in the program, is the code is sometimes hard to follow if you can't make connections to the logic of the individual assignment and class content.

The program does compile!

My changes mostly included formatting and syntax, as the app is still relatively barebones. Some efforts were made to be more specific in visibility declarations of class / function declarations, but really were not necessary.

My concepts learned:
New application of Kotlin best practices when naming variables
Using a landing screen on entry to the app
Necessity of visibility modifiers for classes and functions

File: app/src/main/java/edu/uc/app/parkfinder/RetrofitClientInstance.kt (hunk 1 above)
Comment (discospiff) [reply app/src/main/java/edu/uc/app/parkfinder/RetrofitClientInstance.kt:9]: Usually we only keep the base URL (domain) in the RetrofitClientInstance, and apply the full path in the DAO.   Otherwise, we'd need a BASE_URL for every endpoint we call in that domain, which would obviate the purpose of a BASE_URL.  I'd keep this as it were.

File: app/src/main/java/edu/uc/app/parkfinder/dao/IParkDAO.kt (hunk 2 above)
Comment (discospiff) [reply app/src/main/java/edu/uc/app/parkfinder/dao/IParkDAO.kt:9]: See above.

File: app/src/main/java/edu/uc/app/parkfinder/dto/Park.kt (hunk 3 above)
Comment (discospiff) [reply app/src/main/java/edu/uc/app/parkfinder/dto/Park.kt:5]: Formatting is a user preference, and can be set by the IDE automatically... so I tend not to change a user's preference.  It's safe to leave this as it were.

**Comment by discospiff** [comment]
From this branch:
I wouldn't mere it, though one takeaway: it's a good idea to double check access modifiers on functions and methods.
